src/types/redis_stream_base.h

```diff
@@ -167,12 +174,21 @@ struct StreamConsumerGroupMetadata {
   StreamEntryID last_delivered_id;
   int64_t entries_read = 0;
   uint64_t lag = 0;
+  StreamEntryID max_pending_entry_id;
 };
 
 struct StreamConsumerMetadata {
   uint64_t pending_number = 0;
   uint64_t last_idle;
   uint64_t last_active;
+  StreamEntryID min_pending_entry_id;
+  StreamEntryID max_pending_entry_id;
```

Comment:
Hi @Yangsx-1, currently I only record `min_entry_id/max_entry_id` in stream consumer metadata, because If put every every entry in metadata, it will make the metadata very big. But I am not sure the current design is efficient when I want to read entries in PEL, I have to iterate the elements range from `min_entry_id` and `max_entry_id`, maybe it will be too slow. Do you have any suggestions about this?

Reply:
Yes, it is a little inefficient. But only XPENDING and XINFO commands need to iterate the whole elements, and XACK, XCLAIM will only have o(1) complexity in this design, I think it's ok for now. @LiuYuHui